core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/internal/CDOBasedRepository.java

```diff
@@ -348,5 +363,81 @@ public void handleCommitInfo(CDOCommitInfo commitInfo) {
 			notification.publish(events());
         }
 	}
-	
+
+	@Override
+	public boolean isConsistent(IBranchPath branch) {
+		List<CDOCommitInfo> cdoCommitInfos = getCDOCommitInfos(branch);
+		CommitInfos indexCommitInfos = getIndexCommitInfos(branch);
+		
+		boolean emptyDb = cdoCommitInfos.isEmpty();
+		boolean emptyIndex = indexCommitInfos.getItems().isEmpty();
+
+		if (emptyDb && emptyIndex) {
+			return true; // empty dataset
+		}
+		
+		if (emptyDb ^ emptyIndex) {
+			LOG.error("{} is in inconsistent state. CDO {} but INDEX {}", getCdoRepository().getRepositoryName(), contentMessage(emptyDb), contentMessage(emptyIndex));
+			return false; // either CDO or index was deleted but not the other.
+		}
+		
+		return !hasInvalidCDOTimeStamps(cdoCommitInfos, getLast(indexCommitInfos).getTimeStamp(), branch);
+	}
+
+	private String contentMessage(boolean empty) {
+		return empty ? "IS EMPTY" : "HAS CONTENT";
+	}
+
+	private boolean hasInvalidCDOTimeStamps(List<CDOCommitInfo> immutableCdoCommitInfosList, long lastIndexCommitTimestamp, IBranchPath branch) {
+		// expect all the commit infos to be invalid
+		List<CDOCommitInfo> problematicCommitInfos = newArrayList(immutableCdoCommitInfosList);
+		Iterator<CDOCommitInfo> cdoCommitInfosIterator = problematicCommitInfos.iterator();
+		while (cdoCommitInfosIterator.hasNext()) {
+			CDOCommitInfo cdoCommitInfo = cdoCommitInfosIterator.next();
+			
+			if (cdoCommitInfo.getTimeStamp() <= lastIndexCommitTimestamp) {
+				//removing valid commits from the problematic list
+				cdoCommitInfosIterator.remove();
+			} 
+		}
+		
+		if (!problematicCommitInfos.isEmpty()) {
```

Comment:
If an older restored DB backup is matched up with a newer index, all the CDO commit infos from the problem list would be consumed; it should still be reported as a problem (ie., that now the index is ahead of the DB).